crates/ibc/src/core/context.rs

```diff
@@ -635,16 +745,18 @@ mod val_exec_ctx {
             }
         }
 
+        // state changes
```

Comment:
Some of the parts in `emit events and logs` and `state changes` sections looks similar and gatherable in second functions (as a later work)

Reply:
Indeed, we can include that in the work to be done with #325

Reply:
Sure 👍